bucketeer2/src/main/kotlin/io/bucketeer/sdk/android/internal/scheduler/TaskScheduler.kt

```diff
@@ -0,0 +1,31 @@
+package io.bucketeer.sdk.android.internal.scheduler
+
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import io.bucketeer.sdk.android.internal.di.Component
+import java.util.concurrent.ScheduledExecutorService
+
+internal class TaskScheduler(
+  component: Component,
+  executor: ScheduledExecutorService,
+) : DefaultLifecycleObserver {
+
+  private val foregroundSchedulers: List<ScheduledTask> = listOf(
+    EvaluationForegroundTask(component, executor),
+    EventForegroundTask(component, executor),
+  )
+
+  // app start or back to foreground
+  override fun onStart(owner: LifecycleOwner) {
+    foregroundSchedulers.forEach { it.start() }
+
+    // TODO: stop background tasks
```

Comment:
I'm going to implement background polling in the following PRs

Reply:
but we need to save BKTUser & BKTConfig to somewhere first(SharedPreferences maybe?)